medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -12,40 +15,62 @@ use tokio::time::delay_for;
 
 #[tracing::instrument]
 pub async fn delete_service_worker(
-    storage_container_name: String,
+    time_to_wait: Duration,
+    state_manager: StateManager,
 ) -> Result<String, anyhow::Error> {
+    tracing::info!("Starting delete service worker");
     let mut delete_client = delete_factory().await.map_err(|e| {
         tracing::error!("{:?}", e);
-        anyhow!("Couldn't create the delete queue client")
+        anyhow!("Couldn't delete the Delete Queue")
     })?;
 
     loop {
-        let retrieved_result: Result<RetrievedMessage<DeleteMessage>, RetrieveFromQueueError> =
-            delete_client.receive().await;
-
-        match retrieved_result {
-            Ok(retrieval) => {
-                let message = retrieval.message.clone();
-                tracing::info!("{:?} message receive!", message);
+        match try_process_from_queue(&mut delete_client).await {
+            Ok(status) => match status {
+                FileProcessStatus::Success(job_id) => {
+                    let _ = state_manager.set_status(job_id, JobStatus::Done).await?;
+                }
+                FileProcessStatus::NothingToProcess => {}
+            },
+            Err(e) => {
+                tracing::error!("{:?}", e);
+            }
+        }
+        delay_for(time_to_wait).await;
+    }
+}
 
-                let blob_name = get_blob_name_from_content_id(&message.document_content_id).await?;
-                delete_blob(&storage_container_name, &blob_name)
-                    .await
-                    .map_err(|e| {
-                        tracing::error!("{:?}", e);
-                        anyhow!("Couldn't delete blob {}", &blob_name)
-                    })?;
-                retrieval.remove().await?;
+async fn try_process_from_queue(
+    delete_client: &mut DocIndexUpdaterQueue,
+) -> Result<FileProcessStatus, anyhow::Error> {
+    tracing::info!("Checking for delete messages");
+    let retrieved_result: Result<RetrievedMessage<DeleteMessage>, RetrieveFromQueueError> =
+        delete_client.receive().await;
 
-                // TODO: Notify state manager
-            }
-            Err(azure_error) => tracing::warn!("Azure error! {:?}", azure_error),
-        };
+    if let Ok(retrieval) = retrieved_result {
+        let message = retrieval.message.clone();
+        tracing::info!("{:?} message receive!", message);
 
-        delay_for(Duration::from_secs(10)).await;
+        let storage_container_name = std::env::var("STORAGE_CONTAINER")?;
```

Comment:
I think you're better off using the `env!` macro here, which will insert the environment variable at compile time, if you don't want to pass this variable down.

Reply:
I imagine we'd like to be able to run this against a test storage container though, so it would need to be read at run time.

Reply:
Would you want to change the environment variable without rebuilding? The cost of querying the env variable per message is probably minimal, knowing Rust, so probably doesn't matter how you query this, anyhow!

Reply:
Or would it be better to read this once, higher up the stack, and pass it to the create and delete workers?

Reply:
if we take these at compile time, we'd have to rebuild Docker images to use different values, and we'd have different Docker images in different environments